### PR TITLE
User authentication with OmniAuth facebook

### PR DESCRIPTION
**Pull request by shsteindorsson**
User authentication with OmniAuth facebook
=================================
[PT-link](https://www.pivotaltracker.com/story/show/168988464)
## PR content
In this PR we are adding a feature where the user can log in with Facebook credentials.

## This PR contains:
- The `omniauth-facebook` gem
- A feature test
- Add controller for the Omniauth callback
- Add styling to accommodate Facebook login button
- Add OmniAuth provider authentication to user model
- Add credentials for Facebook OmniAuth to devise initializer
- Add OmniAuth callbacks to Routes
- Add mock information for test

## What have we learned?
- Using third party authentication with OmniAuth
- Stubbing feature tests
- Working with Master.key

## Screenshot
Screenshot 2019-10-11 at 11 34 21

### CODE REVIEW

**Review by miyeaier — APPROVED** [review]

File: features/oauth_facebook.feature [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Feature: Authenticating on prestige worldwide with facebook. 
+  As a user
+	In order to simplify the sign up/login process
```
Comment (oliverochman) [inline features/oauth_facebook.feature]: the indentation is a mess here

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: features/oauth_facebook.feature (hunk 1 above)
Comment (shsteindorsson) [reply features/oauth_facebook.feature]: Agreed, but it doesn't look like that on our machines

Comment (DIzosimov) [reply features/oauth_facebook.feature]: Screenshot 2019-10-11 at 15 28 22

Comment (shsteindorsson) [reply features/oauth_facebook.feature]: Fixed

**Pull request closed without merging**